mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/MongoDBDialect.java

```diff
@@ -191,6 +191,8 @@ public Tuple getTuple(EntityKey key, TupleContext tupleContext) {
 			for ( int i = 0; i < searchObjects.length; i++ ) {
 				if ( dbObject.get( ID_FIELDNAME ).equals( searchObjects[i] ) ) {
 					tuples.set( i, createTuple( keys[i], tupleContext, dbObject ) );
```

Comment:
> I think rather than using a Map I'd see if one can have the objects returned in the order we need them?

Yes, sure, that'd be much better and was in fact my original idea as well. It seems not possible with the current datastore versions, though. E.g. MongoDB does not obey the requested order for a $IN predicate.